### ISSUE TITLE
User Story: Provision and Setup Back-End Development Environment

### ISSUE THREAD

**Issue by pdxdiver** [hackoregon/hack-university-data-architecture#4]
As a developer, I want to be able to be up and running (full working environment) in less than 1 hour

**Comment by znmeb** [comment]
Question: are we going with Docker this time around? Sticking with Ansible?

**Comment by pdxdiver** [comment]
@znmeb, I think this is TBD and ultimately what this class is meant to decide.

**Comment by znmeb** [comment]
As long as it can be made to work on my laptop I'm fine with anything. ;-)

**Comment by pdxdiver** [comment]
@znmeb, what do you think the lowest common denominator laptop across hacko/hacku teams? Windows/Linux/Mac are all represented, but at some point we can't realistically support a 2/4GB netbook and get good performance for local development. Should we draw the line somewhere?

**Comment by znmeb** [comment]
@pdxdiver 
1. Mine is 8 GB but I'm guessing there are lots of 4 GB ones out there. Whatever the minimum is for Docker for Windows / Docker for Mac should work. It should be 64 bits and should have the virtualization assists in the hardware and enabled in the BIOS. 2 GB is probably a non-starter. 
2. It should have the most recent supported OS. Windows 10, the latest MacOS, etc. If it's Linux, it should be the same version we're using in the servers, which I suspect is going to be Ubuntu 16.04.x LTS ("Xenial Xerus"). Frequent mental context switching is a huge drain on productivity. Mine's running Windows 10 / Fedora 25 beta, but I can easily switch when we start building stuff.
3. Browser testing: I'll defer to the front-end folks for this one but I'm noting an awful lot of sites out there work better with Chrome than Firefox after years of Google's developer lock-in.

**Comment by znmeb** [comment]
Here's the Docker for Mac requirements (https://docs.docker.com/docker-for-mac/)

System Requirements: Docker for Mac will launch only if all these requirements are met.
- Mac must be a 2010 or newer model, with Intel’s hardware support for memory management unit (MMU) virtualization; i.e., Extended Page Tables (EPT)
- OS X 10.10.3 Yosemite or newer
- At least 4GB of RAM

VirtualBox prior to version 4.3.30 must NOT be installed (it is incompatible with Docker for Mac)

Note: If your system does not satisfy these requirements, you can install Docker Toolbox, which uses Oracle Virtual Box instead of HyperKit.

**Comment by znmeb** [comment]
And Docker for Windows (https://docs.docker.com/docker-for-windows/). Note that Windows 10 _Home_ won't work - you need at least Windows 10 Pro. Otherwise you have to drop back to the version that uses VirtualBox.

What to know before you install
- README FIRST for Docker Toolbox and Docker Machine users: Docker for Windows requires Microsoft Hyper-V to run. After Hyper-V is enabled, VirtualBox will no longer work, but any VirtualBox VM images will remain. VirtualBox VMs created with docker-machine (including the default one typically created during Toolbox install) will no longer start. These VMs cannot be used side-by-side with Docker for Windows. However, you can still use docker-machine to manage remote VMs.
- The current version of Docker for Windows runs on 64bit Windows 10 Pro, Enterprise and Education (1511 November update, Build 10586 or later). In the future we will support more versions of Windows 10.
- Containers and images created with Docker for Windows are shared between all user accounts on machines where it is installed. This is because all Windows accounts will use the same VM to build and run containers. In the future, Docker for Windows will better isolate user content.
- The Hyper-V package must be enabled for Docker for Windows to work. The Docker for Windows installer will enable it for you, if needed. (This requires a reboot).
- Note: If your system does not satisfy these requirements, you can install Docker Toolbox, which uses Oracle Virtual Box instead of Hyper-V.
- What the install includes: The installation provides Docker Engine, Docker CLI client, Docker Compose, and Docker Machine.

**Comment by natflow** [comment]
I propose moving this conversation to a dedicated issue, which I've just created for the occasion: #11.

**Comment by pdxdiver** [comment]
Good idea , but I think we need to keep the original user story which is meant to sketch out the onboarding and provisioning process
